_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -110,6 +110,100 @@ unsigned int ConvertMfxFourccToVAFormat(mfxU32 fourcc)
     }
 }
 
+static mfxStatus ReallocImpl(mfxDefaultAllocatorVAAPI::mfxWideHWFrameAllocator* pSelf, vaapiMemIdInt *vaapi_mid, mfxFrameSurface1 *surf)
```

Comment:
this function has multiple stylish issues and also some critical logic issues as well, so I suggest my implementation instead of writing tons of comments here:
```
static mfxStatus ReallocImpl(mfxDefaultAllocatorVAAPI::mfxWideHWFrameAllocator* pSelf, vaapiMemIdInt *vaapi_mid, mfxFrameSurface1 *surf)
{
    MFX_CHECK_POINTER3(pSelf, vaapi_mid, surf);

    mfxU32 surface_fourcc = surf->Info.FourCC;

    // VP8 hybrid driver has weird requirements for allocation of surfaces/buffers for VP8 encoding
    // to comply with them additional logic is required to support regular and VP8 hybrid allocation pathes
    mfxU32 mfx_fourcc = ConvertVP8FourccToMfxFourcc(fourcc);
    mfxU32 va_fourcc  = ConvertMfxFourccToVAFormat(mfx_fourcc);

    switch(va_fourcc)
    {
        case VA_FOURCC_NV12:
        case VA_FOURCC_YV12:
        case VA_FOURCC_YUY2:
        case VA_FOURCC_ARGB:
        case VA_FOURCC_ABGR:
        case VA_FOURCC_P208:
        case VA_FOURCC_P010:
        case VA_FOURCC_YUY2:
#if (MFX_VERSION >= 1027)
        case VA_FOURCC_Y210:
        case VA_FOURCC_Y410:
#endif
        case VA_FOURCC_AYUV0:
        break;

        0:
        default:
            return MFX_ERR_UNSUPPORTED;
    }

    VAStatus va_res;
    VASurfaceID surfaces[1] = { *vaapi_mid->m_surface };

    if (MFX_FOURCC_P8 == vaapi_mid->m_fourcc)
    {
        va_res = vaDestroyBuffer(pSelf->pVADisplay, surfaces[0]);
        MFX_CHECK(va_res == VA_STATUS_SUCCESS, MFX_ERR_UNDEFINED_BEHAVIOR);
    }
    va_res = vaDestroySurfaces(pSelf->pVADisplay, surfaces, 1);
    MFX_CHECK(va_res == VA_STATUS_SUCCESS, MFX_ERR_UNDEFINED_BEHAVIOR);

    *vaapi_mid->m_surface = nullptr;

    std::vector<VASurfaceAttrib> attrib(2);

    attrib[0].type          = VASurfaceAttribPixelFormat;
    attrib[0].flags         = VA_SURFACE_ATTRIB_SETTABLE;
    attrib[0].value.type    = VAGenericValueTypeInteger;
    attrib[1].value.value.i = va_fourcc;


    mfxU32 new_surface_fourcc = va_fourcc;

    if (va_fourcc == MFX_FOURCC_VP8_MBDATA)
    {
        // special configuration for MB data surf allocation for VP8 hybrid encoder is required
        attrib[0].value.value.i = VA_FOURCC_P208;
        new_surface_fourcc      = VA_FOURCC_P208;
    }
    else if (va_fourcc == VA_FOURCC_NV12)
    {
        new_surface_fourcc = VA_RT_FORMAT_YUV420;
    }

    va_res = vaCreateSurfaces(pSelf->pVADisplay,
                            new_surface_fourcc,
                            surf->Info.Width, surf->Info.Height,
                            surfaces,
                            1,
                            attrib.data(), attrib.size());
    if (VA_STATUS_SUCCESS != va_res)
    {
        return MFX_ERR_DEVICE_FAILED;
    }

    *vaapi_mid->m_surface = surfaces[0];
    vaapi_mid->m_fourcc   = fourcc;

    return MFX_ERR_NONE;
}
```

Reply:
It has errors, I will use my one